com.unity.shadergraph/Editor/Generation/Processors/PropertyCollector.cs

```diff
@@ -14,26 +16,186 @@ public struct TextureInfo
             public bool modifiable;
         }
 
-        public readonly List<AbstractShaderProperty> properties = new List<AbstractShaderProperty>();
+        private Target m_Target = null;
+        private UInt64 m_DeclarerFlag = (UInt64)DeclarerFlags._NodeOrGraph;
 
-        public void AddShaderProperty(AbstractShaderProperty chunk)
+        private bool m_ReadOnly;
+        private Target m_HLSLPropertyTarget = null;
+        private List<HLSLProperty> m_HLSLProperties = null;
+
+        enum DeclarerFlags
+        {
+            _NodeOrGraph = (1 << 0),    // property declared by a graph (graph property) or a node (or slot on the node)
+            _Target = (1 << 1),         // all bits starting with this one represent properties added by targets in the Target list
+        }
+
+        private struct Property
+        {
+            public AbstractShaderProperty property;
+            public UInt64 declarerFlags;
```

Comment:
For each property, we also track who declared the property - whether it was a node/graph declared property, or if a specific Target declared it.